DAO.sol

```diff
@@ -607,52 +610,40 @@ contract DAO is DAOInterface, Token, TokenCreation {
             sumOfProposalDeposits -= p.proposalDeposit;
         p.open = false;
     }
-	
-    function soloSplit(
+
+    function withdraw(
         uint _proposalID,
         address _newCurator
     ) noEther onlyTokenholders returns (bool _success) {
-		
-		// Did you already vote on another proposal?
+
+        unblockMe();
+
+        // Did you already vote on another proposal?
         if (blocked[msg.sender] != _proposalID && blocked[msg.sender] != 0)  {
```

Comment:
I would also add the following logic near the top of this withdraw() function so that we don't have to execute any of the remaining code in the function:

```
if( balances[msg.sender] <= 0)
    throw;
```

May also want to make the same logic in `splitDAO()`.